src/main/java/org/red5/io/amf/Input.java

```diff
@@ -474,6 +477,8 @@ protected Object readBean(Object bean) {
             }
             if (hasMoreProperties()) {
                 skipPropertySeparator();
+            } else {
+            	break; //hasMoreProperties == false, position moved to +3
```

Comment:
The advance of 3 positions may be cascading, this may be the root of the decoding issue.